server/tapeuse.go

```diff
@@ -0,0 +1,64 @@
+package server
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+// A FixityDB tracks the information the fixity service needs to know what
+
+func (s *RESTServer) EnableTapeUse() {
+	log.Println("Enabling Bendo Tape Use")
+	s.TapeUsed = true
+	s.Items.SetUseStore(true)
+}
+
+func (s *RESTServer) DisableTapeUse() {
+	log.Println("Disabling Bendo Tape Use")
+	s.TapeUsed = false
+	s.Items.SetUseStore(false)
+}
```

Comment:
Maybe I'm missing something here. Since Store.useStore is private,
s.Items.useStore will give an error about trying to access a private ( I
think go calls it it 'unexported' ) field. So I made a Public accessor.

On Thu, Aug 25, 2016 at 12:40 PM, Don Brower <notifications@github.com>
wrote:

> In server/tapeuse.go
> <https://github.com/ndlib/bendo/pull/136#discussion_r76279637>:
>
> > +	"github.com/julienschmidt/httprouter"
> > +)
> > +
> > +// A FixityDB tracks the information the fixity service needs to know what
> > +
> > +func (s *RESTServer) EnableTapeUse() {
> > +	log.Println("Enabling Bendo Tape Use")
> > +	s.TapeUsed = true
> > +	s.Items.SetUseStore(true)
> > +}
> > +
> > +func (s *RESTServer) DisableTapeUse() {
> > +	log.Println("Disabling Bendo Tape Use")
> > +	s.TapeUsed = false
> > +	s.Items.SetUseStore(false)
> > +}
>
> For SetUseStore there is a parameter to toggle on/off. and here there are
> two functions EnableTapeUse and DisableTapeUse...is there a reason these
> are handled differently?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ndlib/bendo/pull/136/files/116b45889c785d5598950eba80d6730d33f98945#r76279637>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADNGTH9a3kACF79r8tZ5fKdLeeXHLnbnks5qjcWUgaJpZM4JsckT>
> .
>